knife/spec/unit/knife/bootstrap_spec.rb

```diff
@@ -1431,19 +1432,29 @@ def configure_gem_files
             knife.config[:connection_password] = "opscode"
           end
           it "includes :connection_password value in a sudo-enabled configuration" do
-            expect(knife.sudo_opts).to eq({
+            expect(knife.sudo_opts(sudo_pass)).to eq({
               sudo: true,
               sudo_password: "opscode",
             })
           end
+
+          context "when sudo_pass is passed" do
+            let(:sudo_pass) { "progress" }
+            it "includes :connection_password value in a sudo-enabled configuration" do
+              expect(knife.sudo_opts(sudo_pass)).to eq({
+                sudo: true,
+                sudo_password: "progress",
+              })
+            end
```

Comment:
This is failing (I know, I suggested it, sorry), because we prefer `connection_password` over sudo_pass (though I'm not sure we should). That said we need to test all combinations:

* connection_password is set, sudo_pass is not set
* connection_password and sudo_pass are both set
* connection_password is not set and sudo_pass is set
* neither are set (probably not useful, but we should know what we expect to happen in this case - raise an error? Nothing? return nil?)

Reply:
Updated the test-cases with different combinations.